### DISCUSSION TITLE
run osxfuse and encfs on catalina (10.15)?

### DISCUSSION THREAD

**Discussion by ouboub** [Homebrew/discussions#1240] [Everyday usage]
Hi 

I started to use a macbook air running catalina (10.15) and installed
first of all fink. 

It took some time to install encfs and osxfuse (currently version 2.7)
however when I want to use encfs on a specific directory I obtain 

,----
| EncFS Password: 
| /opt/sw/lib/osxfusefs.fs/Support/osxfusefs.kext failed to load -
| (libkern/kext) not loadable (reason unspecified); check the
| system/kernel logs for errors or try kextutil(8).
| the OSXFUSE file system is not available (71)
| fuse failed.  Common problems:
|  - fuse kernel module not installed (modprobe fuse)
|  - invalid options -- see usage message
| EncFS Password: 
| 
`----

According to the fink team this is (partially) caused by the fact that
10.15 locked down a lot of system features, including kernel extensions.

So my question is, does encfs and osxfuse work in homebrew?

Thanks

Uwe Brauer

**Comment by carlocab** [comment]
The short answer is no. The slightly longer answer is that you _might_ be able to make it work, but it requires some work.

I'd try installing the `osxfuse` cask and then using `brew extract` to extract `encfs` to your own tap. `encfs` was recently disabled in Homebrew core because of the `osxfuse` requirement.

**Comment by ouboub** [comment]
>>> "CC" == Carlo Cabrera ***@***.***> writes:

> The short answer is no. The slightly longer answer is that you _might_
> be able to make it work, but it requires some work.

I was afraid of that. Since I have a lot of encrypted folders on my
linux machine that I would like to use on my Mac, I am desperate.

Could you outline what more or less should be done, that blocking of
kernel extensions looks serious.

**Comment by carlocab** [comment]
Actually, it looks like @gromgit has already done the work here and is kind enough to host it in a public tap: https://github.com/gromgit/homebrew-fuse

To use it, try
```
brew install --cask gromgit/fuse/macfuse
brew install gromgit/fuse/encfs
```

**Reply by carlocab** [reply]
Actually you may be able to instead do
```
brew install --cask macfuse
brew install gromgit/fuse/encfs
```
which should install a newer version of macFUSE.

**Comment by ouboub** [comment]
>>> "CC" == Carlo Cabrera ***@***.***> writes:


> Actually, it looks like @gromgit has already done the work here and is
> kind enough to host it in a public tap:
> https://github.com/gromgit/homebrew-fuse

> To use it, try
> ```
> brew install --cask gromgit/fuse/macfuse
> brew install gromgit/fuse/encfs
> ```
Thanks I will try to contact him to see what is the secret there.

**Reply by carlocab** [reply]
I believe the secret is to use the binary from https://osxfuse.github.io, so you don't need to worry about signing the kernel extension.

**Comment by ouboub** [comment]
>>> "CC" == Carlo Cabrera ***@***.***> writes:

> I believe the secret is to use the binary from
> https://osxfuse.github.io, so you don't need to worry about signing
> the kernel extension.

Well I just finished my installation of fink (which provides a recent
mercurial version compiled against python 2.7 that I need for my work,
not sure homebrew provides such an «old» version.

So I would like to see whether that can be ported to fink.

**Reply by carlocab** [reply]
If Fink will let you install a pre-compiled binary (built by the macFUSE developer), then it should be straightforward to port it. I suspect that's something Fink doesn't do, though.

**Comment by ouboub** [comment]
>>> "CC" == Carlo Cabrera ***@***.***> writes:


> If Fink will let you install a pre-compiled binary (built by the
> macFUSE developer), then it should be straightforward to port it. I
> suspect that's something Fink doesn't do, though.

Right, I also think that it would not allow the installation of binaries
not compiled by fink itself.

That is why I would like to know who the source code was modified so
that the compiled code allows the use of osxfuse avoiding the kernel
blocking issues with 10.15.

**Reply by carlocab** [reply]
I'm not sure I understand the question. The source code was modified and compiled by the macFUSE developer. It was made closed source a few versions ago, and the versions that are still open source no longer work on new-ish versions of macOS. I believe it's partly related to the requirement that kernel extensions must now be signed with an Apple Developer Certificate.

The binaries compiled by the developer are signed, so they work on newer versions of macOS. However, this probably isn't the entire story: there probably were necessary changes to the source to make them work on newer macOS versions. We don't know what those changes are, because macFUSE is no longer open source software.

**Comment by ouboub** [comment]
>>> "CC" == Carlo Cabrera ***@***.***> writes:

> I'm not sure I understand the question. The source code was modified
> and compiled by the macFUSE developer. It was made closed source a few
> versions ago, and the versions that are still open source no longer
> work on new-ish versions of macOS. I believe it's partly related to
> the requirement that kernel extensions must now be signed with an
> Apple Developer Certificate.

Oh, ok, this I did not know. 

> The binaries compiled by the developer are signed, so they work on
> newer versions of macOS. However, this probably isn't the entire
> story: there probably were necessary changes to the source to make
> them work on newer macOS versions. We don't know what those changes
> are, because macFUSE is no longer open source software.

I will talk to the fink guys but this looks really bad. As longer as I
use MacOS (the recent versions) as more I not, it distances itself from
the Unix world, that is too bad.

**Reply by carlocab** [reply]
Sure. You can read this for more background: https://github.com/osxfuse/osxfuse/issues/590
